db/skipped_sequence.go

```diff
@@ -226,78 +226,82 @@ func (s *SkippedSequenceSlice) _removeSeqRange(ctx context.Context, startSeq, en
 	// if startSeq and endSeq are in different elements there is at least one sequence between these values not in the list
 	if !found {
 		base.DebugfCtx(ctx, base.KeyCache, "sequence range %d to %d specified has sequences in that are not present in skipped list", startSeq, endSeq)
-		return base.ErrSkippedSequencesMissing
+		return false, base.ErrSkippedSequencesMissing
 	}
 	// put this below a check for !found to avoid out of bound error
 	rangeElem := s.list[startIndex]
 	if endSeq > rangeElem.getLastSeq() {
 		base.DebugfCtx(ctx, base.KeyCache, "sequence range %d to %d specified has sequences in that are not present in skipped list", startSeq, endSeq)
-		return base.ErrSkippedSequencesMissing
+		return false, base.ErrSkippedSequencesMissing
 	}
 
 	// handle sequence range removal
 	// update number of sequences currently in slice stat
 	numCurrSkippedSequences := (endSeq - startSeq) + 1
 	s.NumCurrentSkippedSequences -= int64(numCurrSkippedSequences)
 
+	isEmpty := s.NumCurrentSkippedSequences == 0
+
 	// if single sequence element
 	if rangeElem.getStartSeq() == startSeq && rangeElem.getLastSeq() == endSeq {
 		// simply remove the element
 		s.list = slices.Delete(s.list, startIndex, startIndex+1)
-		return nil
+		return isEmpty, nil
 	}
 	// check if one of x or y is equal to start or end seq
 	if rangeElem.getStartSeq() == startSeq {
 		rangeElem.setStartSeq(endSeq + 1)
-		return nil
+		return isEmpty, nil
 	}
 	if rangeElem.getLastSeq() == endSeq {
 		rangeElem.setLastSeq(startSeq - 1)
-		return nil
+		return isEmpty, nil
 	}
 	// assume we are removing from middle of the range
 	newElem := NewSkippedSequenceRangeEntryAt(endSeq+1, rangeElem.getLastSeq(), rangeElem.getTimestamp())
 	s._insert(startIndex+1, newElem)
 	rangeElem.setLastSeq(startSeq - 1)
-	return nil
+	return isEmpty, nil
 
 }
 
 // removeSeq will remove a given sequence from the slice if it exists
-func (s *SkippedSequenceSlice) removeSeq(x uint64) error {
+func (s *SkippedSequenceSlice) removeSeq(x uint64) (bool, error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
 	if len(s.list) == 0 {
-		return fmt.Errorf("skipped sequence list is empty, unable to remove sequence %d", x)
+		return true, fmt.Errorf("skipped sequence list is empty, unable to remove sequence %d", x)
```

Comment:
Is this intetnional to return true here?

Reply:
Yeah as the list is empty here